app/controllers/nonsipp/sharesdisposal/SharesDisposalCYAController.scala

```diff
@@ -81,148 +82,159 @@ class SharesDisposalCYAController @Inject()(
         )
         isIndependentValuation = Option.when(howSharesDisposed == Sold)(
-          request.userAnswers.get(IndependentValuationPage(srn, shareIndex, disposalIndex)).get
+          updatedUserAnswers.get(IndependentValuationPage(srn, shareIndex, disposalIndex)).get
         )
-
         // <Rows conditional on Redeemed>
         dateSharesRedeemed = Option.when(howSharesDisposed == Redeemed)(
-          request.userAnswers.get(WhenWereSharesRedeemedPage(srn, shareIndex, disposalIndex)).get
+          updatedUserAnswers.get(WhenWereSharesRedeemedPage(srn, shareIndex, disposalIndex)).get
         )
         numberSharesRedeemed = Option.when(howSharesDisposed == Redeemed)(
-          request.userAnswers.get(HowManySharesRedeemedPage(srn, shareIndex, disposalIndex)).get
+          updatedUserAnswers.get(HowManySharesRedeemedPage(srn, shareIndex, disposalIndex)).get
         )
         considerationSharesRedeemed = Option.when(howSharesDisposed == Redeemed)(
-          request.userAnswers.get(TotalConsiderationSharesRedeemedPage(srn, shareIndex, disposalIndex)).get
+          updatedUserAnswers.get(TotalConsiderationSharesRedeemedPage(srn, shareIndex, disposalIndex)).get
         )
 
-        // Last row
-        sharesStillHeld <- request.userAnswers
+        sharesStillHeld <- updatedUserAnswers
           .get(HowManyDisposalSharesPage(srn, shareIndex, disposalIndex))
-          .getOrRecoverJourney
+          .getOrRecoverJourneyT
 
         schemeName = request.schemeDetails.schemeName
 
-      } yield Ok(
-        view(
-          viewModel(
-            ViewModelParameters(
-              srn,
-              shareIndex,
-              disposalIndex,
-              sharesType,
-              companyName,
-              acquisitionType,
-              acquisitionDate,
-              howSharesDisposed,
-              dateSharesSold,
-              numberSharesSold,
-              considerationSharesSold,
-              buyerIdentity,
-              buyerName,
-              buyerDetails.flatten,
-              buyerReasonNoDetails.flatten,
-              isBuyerConnectedParty,
-              isIndependentValuation,
-              dateSharesRedeemed,
-              numberSharesRedeemed,
-              considerationSharesRedeemed,
-              sharesStillHeld,
-              schemeName,
-              mode
-            ),
-            viewOnlyUpdated = false, // flag is not displayed on this tier
-            optYear = request.year,
-            optCurrentVersion = request.currentVersion,
-            optPreviousVersion = request.previousVersion,
-            compilationOrSubmissionDate = request.userAnswers.get(CompilationOrSubmissionDatePage(srn))
+        disposalAmount = updatedUserAnswers
+          .map(SharesDisposalProgress.all(srn, shareIndex))
+          .count { case (_, progress) => progress.completed }
+
+        _ <- saveService.save(updatedUserAnswers).liftF
```

Comment:
@Arei1997 @westwater @Vamshi-Muniganti Looks that this save broke the change share disposals functionality by clearing out `SharesDisposalCYAPointOfEntry` value.
Do you by any chance remember why did you need to save again all this data at this point on onPageLoadCommon?

Reply:
I just had a look

This save was used so that what was shown on the CYA page reflected the "completed" state accurately of the disposal journey before being rendered by the view. It was also to ensure a persisted state, if i remember correctly before this change, disposals were not all represented to then be recognised to show the correct inset test

Reply:
Hm, I would think that `SharesDisposalProgress` should be already set before this CYA page is hit.